SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs

```diff
@@ -355,8 +356,16 @@ private void ActivateConsoleLog()
         /// <returns></returns>
         public override bool Flush(int millisecondsTimeout)
         {
-            flushBufferTask?.FlushAndSend().GetAwaiter().GetResult();
-            return true;
+            try
+            {
+                var flushTask = flushBufferTask?.FlushAndSend() ?? Task.FromResult<object>(null);
+                return Task.WhenAny(flushTask, Task.Delay(TimeSpan.FromMilliseconds(millisecondsTimeout))).GetAwaiter().GetResult().Id == flushTask.Id;
+            }
+            catch (Exception ex)

```

Comment:
Since we are catching the exceptions here, we could probably remove the `try-catch` block from the `OnClose` method now?

Reply:
Removed the finally-section.